File: src/shared/api/groups.js
import { invoke } from '@tauri-apps/api/core'

// 获取所有分组
export async function getGroups() {
  return await invoke('get_groups')
}

// 添加分组
export async function addGroup(name, icon = 'ti ti-folder', color = '#dc2626') {
  return await invoke('add_group', { name, icon, color })
}

// 更新分组
export async function updateGroup(oldName, newName, newIcon, newColor) {
  return await invoke('update_group', {
    oldName,
    newName,
    newIcon,
    newColor
  })
}

// 删除分组
export async function deleteGroup(name) {
  return await invoke('delete_group', { name })
}

// 更新分组排序
export async function reorderGroups(groupOrders) {
  return await invoke('reorder_groups', { groupOrders })
}

// 移动收藏项到分组
export async function moveFavoriteToGroup(id, groupName) {
  return await invoke('move_quick_text_to_group', { id, groupName })
}

// 从剪贴板添加到分组
export async function addClipboardToGroup(index, groupName) {
  return await invoke('add_clipboard_to_group', { index, groupName })
}

File: src-tauri/src/commands/groups.rs
use crate::services::database::{get_all_groups, add_group as db_add_group, update_group as db_update_group, delete_group as db_delete_group, reorder_groups as db_reorder_groups, GroupInfo};

// 获取所有分组
#[tauri::command]
pub fn get_groups() -> Result<Vec<GroupInfo>, String> {
    get_all_groups()
}

// 添加分组
#[tauri::command]
pub fn add_group(name: String, icon: String, color: String) -> Result<GroupInfo, String> {
    let result = db_add_group(name, icon, color);
    if result.is_ok() {
        notify_lan_change("groups");
    }
    result
}

// 更新分组
#[tauri::command]
pub fn update_group(old_name: String, new_name: String, new_icon: String, new_color: String) -> Result<GroupInfo, String> {
    let result = db_update_group(old_name, new_name, new_icon, new_color);
    if result.is_ok() {
        notify_lan_change("groups");
    }
    result
}

// 删除分组
#[tauri::command]
pub fn delete_group(name: String) -> Result<(), String> {
    let result = db_delete_group(name);
    if result.is_ok() {
        notify_lan_change("groups");
    }
    result
}

// 更新分组排序
#[tauri::command]
pub fn reorder_groups(group_orders: Vec<(String, i32)>) -> Result<(), String> {
    let result = db_reorder_groups(group_orders);
    if result.is_ok() {
        notify_lan_change("groups");
    }
    result
}

fn notify_lan_change(reason: &'static str) {
    if let Some(app) = crate::services::clipboard::get_app_handle() {
        crate::services::sync_transfer::lan_notify_local_change(app, reason);
    }
}

